src/Umbraco.Web.UI.Client/src/common/services/tinymce.service.js

```diff
@@ -156,6 +156,85 @@ function tinyMceService($rootScope, $q, imageHelper, $locale, $http, $timeout, s
         }
     }
 
+    function uploadImageHandler(blobInfo, success, failure, progress){
+
+        //TODO: Worth refactoring to Angular $http calls
+        //Rather than XHR??
+
+        let xhr, formData;
+
+        xhr = new XMLHttpRequest();
+        xhr.open('POST', Umbraco.Sys.ServerVariables.umbracoUrls.tinyMceApiBaseUrl + 'UploadImage');
+
+        xhr.upload.onprogress = function (e) {
+            progress(e.loaded / e.total * 100);
+        };
+
+        xhr.onerror = function () {
+            failure('Image upload failed due to a XHR Transport error. Code: ' + xhr.status);
+        };
+
+        xhr.onload = function () {
+            let json;
+
+            if (xhr.status < 200 || xhr.status >= 300) {
+                failure('HTTP Error: ' + xhr.status);
+                return;
+            }
+
+            json = JSON.parse(xhr.responseText);
+
+            if (!json || typeof json.location !== 'string') {
+                failure('Invalid JSON: ' + xhr.responseText);
+                return;
+            }
+
+            // Put UDI into localstorage (used to update the img with data-udi later on)
+            localStorage.setItem(`tinymce__${json.location}`, json.udi);
+
+            success(json.location);
+        };
+
+        formData = new FormData();
+        formData.append('file', blobInfo.blob(), blobInfo.blob().name);
+
+        //TODO: Send Media Parent ID from config
+        //TODO: How will each RTE pass in this value to this function?!
+        formData.append('mediaParent', -1);
```

Comment:
Maybe we can just use `Umbraco.Sys.ServerVariables.umbracoSettings.mediaPath` like you use other settings, directly from here?

Reply:
@bergmania this will be set on each on Property Editor config and not global to all RTEs
It was just a noisy comment whilst I was working, so feel free to ignore it